src/compiler/parser.ts

```diff
@@ -8617,8 +8617,19 @@ namespace Parser {
 
         // Parses out a JSDoc type expression.
         export function parseJSDocTypeExpression(mayOmitBraces?: boolean): JSDocTypeExpression {
-            const pos = getNodePos();
+            let pos = getNodePos();
             const hasBrace = (mayOmitBraces ? parseOptional : parseExpected)(SyntaxKind.OpenBraceToken);
+            if (token() === SyntaxKind.AtToken) {
+                    // skip @link since it's a common mistake
+                    if(tokenIsIdentifierOrKeyword(nextTokenJSDoc())) {

```

Comment:
```suggestion
                    if (tokenIsIdentifierOrKeyword(nextTokenJSDoc())) {
```

"dprint when"

Reply:
This is *after* I made an attempt at formatting. I'm getting pre-emptively lazy.

Reply:
keep it up, that way everyone will say yes to formatting!